### PR TITLE
Basic Functionality Implemented

### PR DESCRIPTION
**Pull request by starikcetin**
Components:
* `pwc-ibox`
* `pwc-ibox-title`
* `pwc-ibox-content`
* `pwc-ibox-footer`
* `pwc-ibox-tools`

`pwc-ibox-tools` emits:
* `pwcIboxTools_minimizeClicked`
* `pwcIboxTools_closeClicked`

`pwcIboxTools_minimizeClicked` is listened by:
* `pwc-ibox-content`
* `pwc-ibox-footer`

`pwcIboxTools_closeClicked` is listened by:
* `pwc-ibox`

### CODE REVIEW

**Comment by SchemeSonic** [comment]
@WrathChaos @starikcetin lets review and continue

File: src/components.d.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable */
```
Comment (kuraydev) [inline src/components.d.ts]: We do not need these two lines, just remove them

File: src/components/pwc-ibox-content/pwc-ibox-content.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  tag: "pwc-ibox-content",
+  styleUrl: "../pwc-ibox.scss",
+  shadow: false
+})
+export class PWCIboxContentComponent {
+  @Prop() minimized: boolean = false;
+
+  @Listen("pwcIboxTools_minimizeClicked", { target: "window" })
+  minimizedHandler(event: CustomEvent) {
+    console.log(
```
Comment (kuraydev) [inline src/components/pwc-ibox-content/pwc-ibox-content.tsx]: Remove the console.log for production

File: src/components/pwc-ibox-content/pwc-ibox-content.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import { Component, h, Prop, Listen } from "@stencil/core";
+
+@Component({
+  tag: "pwc-ibox-content",
+  styleUrl: "../pwc-ibox.scss",
+  shadow: false
```
Comment (kuraydev) [inline src/components/pwc-ibox-content/pwc-ibox-content.tsx]: Enable the shadowDOM for production

File: src/components/pwc-ibox-footer/pwc-ibox-footer.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import { Component, h, Prop, Listen } from "@stencil/core";
+
+@Component({
+  tag: "pwc-ibox-footer",
+  styleUrl: "../pwc-ibox.scss",
+  shadow: false
```
Comment (kuraydev) [inline src/components/pwc-ibox-footer/pwc-ibox-footer.tsx]: Enable the shadowDOM for production

File: src/components/pwc-ibox/pwc-ibox.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
+import { Component, h, Prop, Listen, Method } from "@stencil/core";
+
+@Component({
+  tag: "pwc-ibox",
+  styleUrl: "../pwc-ibox.scss",
+  shadow: false
```
Comment (kuraydev) [inline src/components/pwc-ibox/pwc-ibox.tsx]: Enable the shadowDOM for production

File: src/components/pwc-ibox/pwc-ibox.tsx [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+})
+export class PWCIboxComponent {
+  @Prop() closed: boolean = false;
+
+  @Listen("pwcIboxTools_closeClicked", { target: "window" })
+  closedHandler(event: CustomEvent) {
+    console.log("pwc-ibox captured pwcIboxTools_closeClicked", event.detail);
```
Comment (kuraydev) [inline src/components/pwc-ibox/pwc-ibox.tsx]: Remove console.log for production

**Comment by starikcetin** [comment]
Fixed them all.

File: src/components.d.ts (hunk 1 above)
Comment (starikcetin) [reply src/components.d.ts]: This file is auto-generated. My changes would get overwritten wouldn't they? (deleted those lines anyway)

**Review by kuraydev — APPROVED** [review]

**Comment by kuraydev** [comment]
Awesome work :) @starikcetin

**Pull request merged**